app/src/main/java/com/example/topmovies/adapter/FavoriteMoviesAdapter.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.topmovies.adapter
+
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import com.example.topmovies.R
+import com.example.topmovies.databinding.ItemLayoutBinding
+import com.example.topmovies.model.Movie
+import com.example.topmovies.unit.*
+
+class FavoriteMoviesAdapter(private val onClickFavoriteMovie: (String) -> Unit) :
+    RecyclerView.Adapter<FavoriteMoviesAdapter.FavoriteMoviesViewHolder>() {
+    private var favoriteMovies = listOf<Movie>()
+
+    fun setFavoriteMovies(movies: List<Movie>) {
+        favoriteMovies = movies
+        notifyItemRangeChanged(0, movies.size)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteMoviesViewHolder {
+        val binding = ItemLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return FavoriteMoviesViewHolder(binding, onClickFavoriteMovie, ::removeMovie)
+    }
+
+    override fun onBindViewHolder(holder: FavoriteMoviesViewHolder, position: Int) =
+        holder.bind(favoriteMovies[position])
+
+    override fun getItemCount() = favoriteMovies.size
+
+    private fun removeMovie(movie: Movie) {
+        val index = favoriteMovies.indexOf(movie)
+        val mutableList = favoriteMovies.toMutableList()
+        mutableList.removeAt(index)
+        favoriteMovies = mutableList.toList()
+        notifyItemRemoved(index)
```

Comment:
too many operations on that list.
Please find a way to optimize this approach.
While deleting one Item you are creating 2 extra lists.